api/data_refinery_api/management/commands/post_downloads_summary.py

```diff
@@ -0,0 +1,130 @@
+                days, len(unique_users), len(unique_ips)
+            )
+        else:
+            fallback_text = "There were no downloads in the last {0} days.".format(days)
+
+        new_users = ""
+        returning_users = ""
+        for email in unique_users:
+            user_annotations = annotation_queryset.filter(dataset__email_address=email)
+            total_downloads = user_annotations.count()
+            unique_locations = list(set(annotation.data["ip"] for annotation in user_annotations))
+            locations = ", ".join(get_ip_location(ip) for ip in unique_locations)
+            is_new_user = DatasetAnnotation.objects.filter(
+                created_at__lt=start_time, dataset__email_address=email
+            )
+            text = "{0} | {1} downloads from {2}\n".format(email, total_downloads, locations)
+            if is_new_user:
+                new_users += text
+            else:
+                returning_users += text
+
+        blocks = [
+            {
+                "type": "section",
+                "text": {"type": "plain_text", "emoji": True, "text": fallback_text},
+            }
+        ]
+        if new_users:
+            blocks.append(
+                {
+                    "type": "section",
+                    "text": {"type": "mrkdwn", "text": "*New users* \n" + new_users,},
+                }
+            )
+        if returning_users:
+            blocks.append(
+                {
+                    "type": "section",
+                    "text": {"type": "mrkdwn", "text": "*Returning users* \n" + returning_users,},
+                }
+            )
+
+        # Post to slack
+        requests.post(
+            settings.ENGAGEMENTBOT_WEBHOOK,
+            json={
+                "username": "EngagementBot",
+                "icon_emoji": ":halal:",
+                "channel": "#" + options["channel"],
+                "text": fallback_text,
+                "blocks": blocks,
+            },
+            headers={"Content-Type": "application/json"},
+            timeout=10,
+        )
+
+
+def should_display_email(email: str) -> bool:
+    """ Returns true if the given email is not associated with the CCDL suers """
+    if not email:
+        return False
+
+    return not (
+        email.startswith("cansav09")
+        or email.startswith("arielsvn")
+        or email.startswith("jaclyn.n.taroni")
+        or email.startswith("kurt.wheeler")
+        or email.startswith("greenescientist")
+        or "@alexslemonade.org" in email
+        or email.startswith("miserlou")
+        or email.startswith("d.prasad")
+        or email is ("daniel.himmelstein@gmail.com")
+        or email is ("dv.prasad991@gmail.com")
+    )
+
+
+def get_ip_location(remote_ip):
+    try:
+        city = requests.get("https://ipapi.co/" + remote_ip + "/json/", timeout=10).json()["city"]
+    except Exception:
+        city = "COULD_NOT_DETERMINE"
```

Comment:
If we can't get the city it might be because the ipapi is down. In this case can we then include the IP address in the slackbot message so that we can try looking it up later?